core/templates/components/summary-tile/story-summary-tile-displayed-chapters-calculator.directive.ts

```diff
@@ -0,0 +1,46 @@
+// Copyright 2020 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Component to be used in the story summary tile to calculate
+ * number of chapters displayed initially.
+ */
+ 
+ /* eslint-disable angular/directive-restrict */
```

Comment:
Also, if it helps with getting this PR in, I'm fine with showing a fixed number of chapters for now at the outset. Maybe get guidance from @mschanteltc about what the exact behaviour should be for the number of chapters shown at the outset -- should it be varying based on height or is it OK for us to approximate this by a fixed number?

Reply:
But how can we decide on a fixed number? Since at 1920 x 1080, around 4 chapters can be fit, which would be higher if the display is 4K, but at 1280 x 720 and lower, even 2 can overflow.

So, if we fix it to a low number, then 'Show More' would be shown for high res displays even though the box is barely filled, but if we fix it higher, the box would overflow regardless of the limit for low res displays.

Considering that an 'expan' functionality exists anyway, maybe fixing it to a high value would be better since overflowing for a low res display wouldn't look too bad compared to showing 'Show More' when the box is barely filled.

Reply:
I think this is fine for now -- let's fix it to the higher value and could you please start an email thread with @mschanteltc (with screenshots if possible) to ask about what the actual behaviour should be?

Reply:
Ok